gobblin-core/src/main/java/org/apache/gobblin/publisher/BaseDataPublisher.java

```diff
@@ -292,6 +296,33 @@ private void publishMultiTaskData(WorkUnitState state, int branchId, Set<Path> w
     publishData(state, branchId, false, writerOutputPathsMoved);
   }
 
+  public boolean submitSingleTaskLineageEvent (WorkUnitState state) {
```

Comment:
Why is this part of `BaseDataPublisher`? Maybe put it in `LineageEvent` instead?

Reply:
@ibuenros This will allow individual publisher decide if it needs to submit lineage event. If we move it to LineageEvent. We will still need a hook in the publisher, and that means for all the task publishers we will always send lineage event even if we don't need to.

Reply:
If a `DataPublisher` should have the ability to submit some event, we can add a function `submitEvent(WorkUnitState)` in the `DataPublisher` class.

Another idea is similar to hive registration and audit count, we can have a `DatasetEventPublisher` which sends events at the dataset level but is not be able to send publisher specific event.